react-native-aztec/android/src/main/java/org/wordpress/mobile/ReactNativeAztec/ReactAztecText.java

```diff
@@ -321,6 +340,50 @@ private boolean onBackspace() {
         return true;
     }
 
+    /**
+     * Handle paste action by retrieving clipboard contents and dispatching a
+     * {@link ReactAztecPasteEvent} with the data
+     *
+     * @param   isPastedAsPlainText boolean indicating whether the paste action chosen was
+     *                         "PASTE AS PLAIN TEXT"
+     *
+     * @return  boolean to indicate that the action was handled (always true)
+     */
+    private boolean onPaste(boolean isPastedAsPlainText) {
+        ClipboardManager clipboardManager = (ClipboardManager) getContext().getSystemService(
+                Context.CLIPBOARD_SERVICE);
+
+        StringBuilder text = new StringBuilder();
+        StringBuilder html = new StringBuilder();
+
+        if (clipboardManager != null && clipboardManager.hasPrimaryClip()) {
+            ClipData clipData = clipboardManager.getPrimaryClip();
+            int itemCount = clipData.getItemCount();
+
+            for (int i = 0; i < itemCount; i++) {
+                Item item = clipData.getItemAt(i);
+                text.append(item.coerceToText(getContext()));
+                if (!isPastedAsPlainText) {
+                    html.append(item.coerceToHtmlText(getContext()));
+                }
+            }
+        }
+
+        // temporarily disable listener during call to toHtml()
+        disableTextChangedListener();
+        String content = toHtml(false);
+        int cursorPositionStart = getSelectionStart();
+        int cursorPositionEnd = getSelectionEnd();
+        enableTextChangedListener();
+        ReactContext reactContext = (ReactContext) getContext();
+        EventDispatcher eventDispatcher = reactContext.getNativeModule(UIManagerModule.class)
+                .getEventDispatcher();
+        eventDispatcher.dispatchEvent(new ReactAztecPasteEvent(getId(), content,
+                cursorPositionStart, cursorPositionEnd, text.toString(), html.toString())
```

Comment:
Just one question on my end.

I see that when `isPastedAsPlainText == false`, we're sending both the text representation and the HTML representation with the event.  Will both be necessary simultaneously in our Javascript code?

The reason I'm asking is that if they're not necessary simultaneously, we can split this into two events (think `ReactAztecPasteEvent` and `ReactAztecPasteHTMLEvent`) to make the interface less ambiguous.

Reply:
Thanks for taking a look @diegoreymendez. Your question is a very good one!
> Will both be necessary simultaneously in our Javascript code?

The Javascript code has existing implementations for handling paste and that code expects both html and plain text data to be present. Splitting the event would clarify the interface, but there is also a benefit to maintaining common abstraction boundaries between the code bases. In this case, because there is already so much code in place on the Javascript side, I think it is best to mirror that implementation.

Reply:
That sounds reasonable @mkevins.